docs/PacketFence_Installation_Guide.asciidoc

```diff
@@ -2440,215 +2440,106 @@ And can be defined using different criteria like:
 
-.EXAMPLE: autoregister the device and assign the role staff to each device where the username is igmout.
-[source,ini,numbered]
-----
-  [igmout]
-  filter = username
-  operator = is
-  value = igmout
+.EXAMPLE_Autoregister_Printer_Scanner will autoregister printers and scanners and add a note.
 
-  [secure]
-  filter = ssid
-  operator = is
-  value = SECURE
+You can have a look in the file `vlan_filters.conf`, there are some examples on how to use and define filters.
 
-  [3:igmout&secure]
-  scope = AutoRegister
-  role = staff
+Advanced LDAP Authentication
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-  [4:igmout&secure]
-  scope = NodeInfoForAutoReg
-  role = staff
-----
+ldapfilter action
+^^^^^^^^^^^^^^^^^
 
-.EXAMPLE: autoregister all Windows devices and assign them the role default.
-[source,ini,numbered]
-----
-  [Windows_AutoReg]
-  filter = fingerbank_info.device_hierarchy_names
-  operator = includes
-  value = Windows
+ldapfilter action override the internal ldap filter packetfence create internally (uid=$username) so you can create a custom filter that match your needs.
 
-  [AUTOREG_WINDOWS:Windows_AutoReg]
-  scope = AutoRegister
-  role = default
+Per example something like this (search for the user and check to see if it's permitted based on some criteria):
 
-  [WINDOWS_ROLE:Windows_devices]
-  scope = NodeInfoForAutoReg
-  role = default
-----
+  (&(|(cn=${radius_request.Stripped-User-Name})(cn=${radius_request.User-Name}))(|(permitWifi=*)(grade=staff)(memberOf=CN=WifiGroup,OU=Security Groups,DC=ad,DC=acme,DC=com)))
 
-.EXAMPLE: filter a MAC address and reject it by assigning the role REJECT.
-[source,ini,numbered]
-----
-  [MAC]
-  filter = node_info.mac
-  operator = match
-  value = aa:bb:cc:dd:ee:ff
+Set_role_on_not_found action
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-  [SSID_Open]
-  filter = ssid
-  operator = is
-  value = Open
+set_role_on_not_found is a way to define a role is the rule doesn't match, let's take the ldapfilter example above.
 
-  [REJECT:MAC&SSID_Open]
-  scope = RegistrationRole
-  role = REJECT
-----
+If we add the action set_role_on_not_found = REJECT so it mean that the device will be rejected if the ldap filter doesn't return anything. (if it match then set_role action will be applied)
 
-.EXAMPLE: autoregister devices using dot1x/EAP over wired or wireless and that are not pending
-[source,ini,numbered]
-----
-  [not_pending]
-  filter = node_info.status
-  operator = is_not
-  value = pending
+role_from_source action
+^^^^^^^^^^^^^^^^^^^^^^^
 
-  [wireless_dot1x]
-  filter = connection_type
-  operator = is
-  value = Wireless-802.11-EAP
+role_from_source will check to see if the ldap attribute exist and will add it in the ldap_attribute context (available in the radius filters)
 
-  [wired_dot1x]
-  filter = connection_type
-  operator = is
-  value = Ethernet-EAP
+So per example if you want to take the value of the ldap attribute customRadius and add it in the radius answer you needs to do the following.
```

Comment:
So per example -> So for example
ldap attribute -> LDAP attribute
radius -> RADIUS
you needs -> you need